### DISCUSSION TITLE
Improve code auto-completion

### DISCUSSION THREAD

**Discussion by CodingOctocat** [godotengine/godot-proposals#7682] [Editor]
The current code completion uses the `Tab` key to submit, and it is proposed to support more keystroke submissions(`Space`, `(`, `[`, `.`):

This avoids frequent use of the `Tab` key.

current behavior：
`pri` -> `Tab` -> `(` => `print()`
improve：
`pri` -> `(` => `print()`

current behavior：
`my_o` -> `Tab` -> `.` -> `prop` -> `Tab` => `my_obj.property1`
improve：
`my_o` -> `.` -> `prop` -> `Space`/`Tab` => `my_obj.property1`

**Comment by Calinou** [comment]
Pressing <kbd>Space</kbd> to accept autocompletion suggestions is pretty uncommon in most editors. You should already be able to add this binding in the Editor Settings' shortcuts tab, but I don't think it should be the default.

**Comment by CodingOctocat** [comment]
@Calinou Maybe I didn't describe it very well, most IDEs (Visual Studio, PyCharm, Rider) support (`Space`, `(`, `[`, `. `), where the IDE submits the candidate and then types the symbols, which is equivalent to merging the submit and symbol operations.

Space are common because programming languages use spaces as separators, as shown below.
1. `pub -> Tab -> Space -> Foo` It can be simplified to： `pub -> Space -> Foo`
2. `myd -> Tab -> . -> A -> Tab -> ( -> 1` It can be simplified to：`myd -> . -> ( -> 1`

![动画2](https://github.com/godotengine/godot-proposals/assets/7220248/1238e36f-e95e-48d0-9371-39913341f733)
